bundles/org.openhab.core/src/main/java/org/openhab/core/util/LightModel.java

```diff
@@ -0,0 +1,638 @@
+ * case above except that the white channel is derived from the lowest of the RGB values and all values are clamped in
+ * the range [0..255]. The '{@link supportsRgbDimming}' changes the behavior in relation to 'HS' versus 'HSB' exactly as
+ * in the case of RGB above</li>
+ *
+ * </ol>
+ * <p>
+ * A typical use case is within in a ThingHandler as follows:
+ *
+ * <pre>
+ * {@code
+ * public class LightModelHandler extends BaseThingHandler {
+ *
+ *     // initialize the light model with default capabilities and parameters
+ *     private final LightModel model = new LightModel();
+ *
+ *     @Override
+ *     public void initialize() {
+ *         // adjust the light model capabilities
+ *         model.configSetSupportsBrightness(true);
+ *         model.configSetSupportsColorTemperature(true);
+ *         model.configSetSupportsColor(true);
+ *
+ *         // adjust the light model configuration parameters
+ *         model.configSetMiredCoolest(153);
+ *         model.configSetMiredWarmest(500);
+ *     }
+ *
+ *     &#x40;Override
+ *     public void handleCommand(ChannelUID channelUID, Command command) {
+ *         // update the model state based on a command from OpenHAB
+ *         model.handleCommand(command);
+ *
+ *         // or if it is a color temperature command
+ *         model.handleColorTemperatureCommand(command);
+ *
+ *         sendBindingSpecificCommandToUpdateRemoteLight(
+ *              .. model.getOnOff() or
+ *              .. model.getBrightness() or
+ *              .. model.getColor() or
+ *              .. model.getColorTemperature() or
+ *              .. model.getColorTemperaturePercent() or
+ *              .. model.getRGBx() or
+ *              .. model.getXY() or
+ *         );
+ *     }
+ *
+ *     // method that sends the updated state data to the remote light
+ *     private void sendBindingSpecificCommandToUpdateRemoteLight(..) {
+ *       // binding specific code
+ *     }
+ *
+ *     // method that receives data from remote light, and updates the model, and then OH
+ *     private void receiveBindingSpecificDataFromRemoteLight(double... receivedData) {
+ *         // update the model state based on the data received from the remote
+ *         model.setBrightness(receivedData[0]);
+ *         model.setRGB(receivedData[1], receivedData[2], receivedData[3]);
+ *         model.setMired(receivedData[4]);
+ *
+ *         // update the OH channels with the new state values
+ *         updateState(onOffChannelUID, model.getOnOff());
+ *         updateState(brightnessChannelUID, model.getBrightness());
+ *         updateState(colorChannelUID, model.getColor());
+ *         updateState(colorTemperatureChannelUID, model.getColorTemperature());
+ *     }
+ * }
+ * }
+ * </pre>
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class LightModel {
+
+    /**
+     * The internal logic is implementated via an instance of this 'hidden' class
```

Comment:
What am I missing here? LightModelAbstractLogicImpl is an abstract class, with no concrete implementations. How are you able to instantiate it in the constructor? And what's the point of it anyway, given that a specific implementation (well-- the abstract version, anyway) is explicitly created in the constructor for this class, with no opportunity for anything else to influence which implementation is used.

I _could_ see this being important for some of my more complicated RGBWW use cases I outlined above, but don't see how it's supposed to fit together yet.

Reply:
Ohhhhhh, sneaky. I still consider myself only mediocre at Java :).